docs/spfx/sharepoint-subscription-edition-component.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Upgrade components in SharePoint Framework development with SharePoint Server Subscription Edition
+description: SharePoint Server Subscription Edition (SE) supports to ungrade individual components within SharePoint Framework when developing client-side web parts.
+ms.date: 09/08/2023
+ms.localizationpriority: high
+---
+
+
+# Upgrade components in SharePoint Framework development with SharePoint Server Subscription Edition
+
+SharePoint Server Subscription Edition supports SharePoint Framework client-side web parts in classic and modern pages, and extensions in modern pages.
+
+When developing a client-side web part, you can only use the specific versions of components supported by the SharePoint Framework to make the web part compatible with your SharePoint Server.
+
+To use more modern versions of the SharePoint Framework (SPFx) components, this article introduces some ways to upgrade individual components without upgrading the major version of SPFx.
+
+> [!NOTE]
+> Because the component-wide upgrades were not made as part of a broader update to SPFx, some manual configuration of the development environment might be completed in order to utilize them.
+
+> [!IMPORTANT]
+> Only SharePoint Server Subscription Edition (SE) supports the component-wide upgrades for the SPFx development since the [Version 23H2](/sharepoint/what-s-new/new-and-improved-features-in-sharepoint-server-subscription-edition-23h2-release#sharepoint-framework-spfx-component-upgrades) feature update. SharePoint Server 2019 and older SharePoint Server products do not support that.
+
+## Upgrade React and Fluent UI React
+
+The SharePoint Framework v1.4.1 natively supports React v15.8 and Fluent UI React ([formerly Office UI Fabric React](https://devblogs.microsoft.com/microsoft365dev/ui-fabric-is-evolving-into-fluent-ui/)) v5.17.
+
+If you want to use React v16 or Fluent UI React v7 in your custom web parts developed for SharePoint Server with the SharePoint Framework v1.5.0, here are the instructions:
+
+1. Modify the "dependencies" and "devDependencies" in the npm project:

```

Comment:
All these manual steps to change the versions of SPFx project can be implemented using a script specifying specific versions to install. These steps don't address the **package-lock.json** file which would also need to be updated if you manually edit these version numbers.

But if you instead used a script, it would incorporate those changes in not just the **package.json** but also the **package-lock.json** file. Furthermore, it's generally accepted guidance to use NPM scripts to upgrading NPM packages, not to manually edit the **package.json** file.

For example, the first bullet point could be replaced with:

```console
npm install react@16.13.1 react-dom@16.13.1 @types/react@16.9.51 @types/react-dom@16.9.8 @microsoft/office-ui-fabric-react-bundle@1.15.0 -SE
npm install @microsoft/sp-build-web@1.9.1 @microsoft/sp-module-interfaces@1.9.1 -DE
```